### DISCUSSION TITLE
webpack latest version 5 does not working node_modules dropping unused Icon for package libary @mdi/js and @mdi/react

### DISCUSSION THREAD

**Discussion by nitinr010** [webpack/webpack#14999] [Q&A]
# Bug report

**What is the current behavior?**
node_modules dropping unused Icon for package libary @mdi/js and @mdi/react

**If the current behavior is a bug, please provide the steps to reproduce.**

I have included "sideEffects": false to  package.json  and webpack.prod.js file but nothing seems to work

Not droping unused icon for below image 

webpack.config.js file

const path = require('path');
const webpack = require('webpack');
const packageConfig = require('./package');
const appConfig = require('./webpack.appConfig');
const getPath = require('lodash/get');

console.log('APP VER: ', packageConfig.version);

const HtmlWebpackPlugin = require('html-webpack-plugin');
const UglifyJsPlugin = require('uglifyjs-webpack-plugin');
const TerserJSPlugin = require('terser-webpack-plugin');
const MiniCssExtractPlugin = require('mini-css-extract-plugin');
const CssMinimizerPlugin = require('css-minimizer-webpack-plugin');
// const OptimizeCSSAssetsPlugin = require('optimize-css-assets-webpack-plugin');
const {CleanWebpackPlugin} = require('clean-webpack-plugin');
const BabelMinifyPlugin = require('babel-minify-webpack-plugin')
const BundleAnalyzerPlugin = require('webpack-bundle-analyzer').BundleAnalyzerPlugin;
// const PreloadWebpackPlugin = require('preload-webpack-plugin');
const TsconfigPathsPlugin = require('tsconfig-paths-webpack-plugin');
const DeadCodePlugin = require('webpack-deadcode-plugin');

module.exports = {
  entry: {
    app: [path.resolve(__dirname, './js/index.jsx'),
      path.resolve(__dirname, './favicon.ico'),
      path.resolve(__dirname, './robots.txt')],
  },
  output: {
    publicPath: '/',
    path: path.resolve(__dirname, './dist'),
    filename: '[name].[contenthash].bundle.js',
    chunkFilename: '[name].[contenthash].bundle.js',
    library: 'mode',
    libraryTarget: 'umd',
  },
  optimization: {
    usedExports: true,
    concatenateModules: false,
    sideEffects: true,
    minimize: true,
    minimizer: [
      `...`,
      new TerserJSPlugin({
        parallel: true,
        terserOptions: {
          ecma: 6,
        },
      }),
      new UglifyJsPlugin(),
      // new BabelMinifyPlugin(),
      new CssMinimizerPlugin(),
    ],
    splitChunks: {
      chunks: 'all',
      minSize: 20000,
      maxSize: 500000,
      cacheGroups: {
        styles: {
          name: 'styles',
          test: /\.(css|scss)$/,
          chunks: 'all',
          enforce: true,
        },
      },
    },
  },
  devServer: {
    contentBase: './',
    publicPath: './dist/',
  },
  resolve: {
    extensions: ['.jsx', '.js', '.ts', '.tsx'],
    plugins: [new TsconfigPathsPlugin({
      extensions: ['.jsx', '.js', '.ts', '.tsx'],
    })],
  },
  devtool: 'source-map',
  mode: 'production',
  module: {
    rules: [
      {
        test: /\.ts(x?)$/,
        exclude: /node_modules/,
        use: [
          {
            loader: 'ts-loader',
          },
        ],
      },
      {
        test: /\.(jsx|js|ts|tsx)$/,
        exclude: /node_modules/,
        loader: 'string-replace-loader',
        options: {
          search: '__CONST__\.([\\w\\d\\.]+)',
          replace: (match, p1) => {
            const replaced = getPath(appConfig, p1);
            if (replaced === undefined) throw new Error(`cannot find replacement for constant value "${p1}"`);
            return JSON.stringify(replaced);
          },
          flags: 'g',
        },
      },
      {
        test: /\.(jsx|js)$/,
        exclude: /node_modules/,
        use: ['babel-loader'],
      },
      {
        test: /\.(sass|scss|css)$/,
        include: [path.resolve(__dirname, '.'), path.resolve(__dirname, '/styles/fonts/nunito')],
        use: [
          {
            loader: MiniCssExtractPlugin.loader,
          },
          'css-loader',
          'postcss-loader',
          {
            loader: 'sass-loader',
            options: {
              implementation: require('sass'),
              additionalData: '@use \'sass:math\'; @import "styles/base";',
              sassOptions: {
                includePaths: [path.resolve(__dirname, '.')],
                outputStyle: 'compressed',
              },
              sourceMap: true,
            },
          },
        ],
      },
      {
        test: /\.(woff(2)?|[ot]tf|eot|svg)([?#]+[a-z0-9]+)?$/i,
        include: [/fonts/],
        type: 'asset/resource',
      },
      {
        test: /\.svg$/,
        exclude: [/fonts/],
        type: 'asset/resource',
      },
      {
        test: /.*/,
        include: [path.resolve(__dirname, './favicon.ico'), path.resolve(__dirname, './robots.txt')],
        type: 'asset/resource',
        generator: {
          filename: '[name][ext]',
        }
      },
      {
        test: /\.(jpe?g|png|gif)$/i,
        type: 'asset/resource',
      },
      // All output '.js' files will have any sourcemaps re-processed by 'source-map-loader'.
      {
        test: /\.js$/,
        exclude: /node_modules/,
        enforce: 'pre',
        loader: 'source-map-loader',
      },
    ],
  },
  plugins: [
    new HtmlWebpackPlugin({
      template: path.resolve(__dirname, '../views/layouts/layout-main_template.ejs'),
      filename: path.resolve(__dirname, '../views/layouts/layout-main.ejs'),
    }),
    // new PreloadWebpackPlugin(),
    new MiniCssExtractPlugin({
      // Options similar to the same options in webpackOptions.output
      // all options are optional
      filename: '[name].[contenthash].bundle.css',
      chunkFilename: '[name].[contenthash].bundle.css',
      ignoreOrder: false, // Enable to remove warnings about conflicting order
    }),
    new DeadCodePlugin({
      patterns: [
      'src/**/*.(js|jsx|sass|scss|css|ts|tsx)',
      ],
      exclude: [
      '**/*.(stories|spec).(js|jsx|ts|tsx)',
      ],
      }),
    new webpack.ProvidePlugin({
      $: 'jquery',
      jQuery: 'jquery',
      'moment': 'moment-timezone',
    }),
    new webpack.DefinePlugin({
      'process.env': {
        appVersion: JSON.stringify(packageConfig.version),
        NODE_ENV: JSON.stringify('production'),
      }
    }),
    new CleanWebpackPlugin(),
    new webpack.optimize.LimitChunkCountPlugin({
      maxChunks: 40
    }),
    // new BundleAnalyzerPlugin(),
    // new UglifyJsPlugin(),
    new BundleAnalyzerPlugin(),
    //new OptimizeCssAssetsPlugin({
    //  assetNameRegExp: /\.css$/g,
    //  cssProcessorOptions: {safe: true, discardComments: {removeAll: true}}
    //})
  ],
};

**What is the expected behavior?**

Below image inside droping unused icon for @mdi/js and @mdi/react
![image](https://user-images.githubusercontent.com/93132632/146308601-679374b2-089a-4a0f-84d2-d6c871e18dee.png)

**Other relevant information:**
webpack version:  5.65.0
Node.js version: 12.14.1
Operating System: linux
Additional tools:

I don't know what is missing in this file please any solution then let me know

**Comment by alexander-akait** [comment]
Please create reproducible test repo, code in the issue is not readable
